### ISSUE TITLE
Understand complexFilter method

### ISSUE THREAD

**Issue by pirmax** [fluent-ffmpeg/node-fluent-ffmpeg#1136]
Someone to help me to understand how complex filters works?

I'd like to use this filter with **complexFilter** method: https://stackoverflow.com/a/30832903/8118350

I don't know how I can cutting this:

`[0:v]scale=ih*16/9:-1,boxblur=luma_radius=min(h\,w)/20:luma_power=1:chroma_radius=min(cw\,ch)/20:chroma_power=1[bg];[bg][0:v]overlay=(W-w)/2:(H-h)/2,crop=h=iw*9/16`

**Comment by all-things-liquid** [comment]
**TL;DR**: You should split the string it using the `;` as delimiter, and pass the resulting array as argument of `complexFilter()`

---
Here is some general info about complex filters if you ever need to use it again:
- Each complex filter command is a combination of `input(s)-filter(s)` (and sometimes `-output(s)`)
  - if no output is specified, then the output will be the one given at the end of the FFmpeg command
  - you don't need to specify an output unless you plan to use it as an input for more filters -> the output of a first combination can be the input of the second one
- `[ ]` indicates input or output information for filters
  - an input with a number like `[0:v]` means we take the first input given with the `-i` option earlier in the command
  - an input with no number like `[bg]` means we take the output of a previous group as input 
- `:` separates filter options
- `,` separates filters
- `;` separates groups of input(s)-filter(s)-output(s)

And a more detailed explanation of the string you want to split, with newlines at each `:`, `,` and `;`:

``` javascript
// First group
  // This is the input for the first batch of filters
  [0:v] 
    scale=
      // These are options for the scale filter
      w=ih*16/9: 
      h=-1,
    boxblur=
      // These are options for the boxblur filter
      luma_radius=min(h\,w)/20:
      luma_power=1:
      chroma_radius=min(cw\,ch)/20:
      chroma_power=1
  // This is the output of [0:v] going through the 2 filters 
  [bg]; 
// Second group
  // These are the 2 inputs required for the overlay filter
  [bg][0:v]
    overlay=
      // These are options for the overlay filter
      x=(W-w)/2:
      y=(H-h)/2,
    crop=
    // These are options for the crop filter
      h=iw*9/16
  // No output is given here as it will use the one from the FFmpeg full command
```

**Comment by utkarshx** [comment]
[
  {
    "filter": "scale",
    "options": "ih*16/9:-1",
    "inputs": "[0:v]",
    "outputs": "[scaled]"
  },
  {
    "filter": "boxblur",
    "options": {
      "luma_radius": "min(h\\,w)/20",
      "luma_power": 1,
      "chroma_radius": "min(cw\\,ch)/20",
      "chroma_power": 1
    },
    "inputs": "[scaled]",
    "outputs": "[bg]"
  },
  {
    "filter": "overlay",
    "options": "(W-w)/2:(H-h)/2",
    "inputs": ["[bg]", "[0:v]"],
    "<EUGPSCoordinates>crop</EUGPSCoordinates>h=iw*9/16"
  }
]
